File: Admin_p/Admin_p.Web/Modules/electroDB/Promotions/PromotionsForm.cs
using Serenity.ComponentModel;
using System;

namespace Admin_p.electroDB.Forms;

[FormScript("electroDB.Promotions")]
[BasedOnRow(typeof(PromotionsRow), CheckNames = true)]
public class PromotionsForm
{
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal DiscountPercentage { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public bool IsActive { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: Admin_p/Admin_p.Web/Modules/ServerTypes/electroDB/PromotionsForm.ts
import { StringEditor, DecimalEditor, DateEditor, BooleanEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface PromotionsForm {
    Name: StringEditor;
    Description: StringEditor;
    DiscountPercentage: DecimalEditor;
    StartDate: DateEditor;
    EndDate: DateEditor;
    IsActive: BooleanEditor;
    CreatedAt: DateEditor;
    UpdatedAt: DateEditor;
}

export class PromotionsForm extends PrefixedContext {
    static readonly formKey = 'electroDB.Promotions';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PromotionsForm.init)  {
            PromotionsForm.init = true;

            var w0 = StringEditor;
            var w1 = DecimalEditor;
            var w2 = DateEditor;
            var w3 = BooleanEditor;

            initFormType(PromotionsForm, [
                'Name', w0,
                'Description', w0,
                'DiscountPercentage', w1,
                'StartDate', w2,
                'EndDate', w2,
                'IsActive', w3,
                'CreatedAt', w2,
                'UpdatedAt', w2
            ]);
        }
    }
}